apps/test/unit/gamelab/GameLabSpriteTest.js

```diff
@@ -300,6 +303,86 @@ describe('GameLabSprite', function () {
+    const ANIMATION_LABEL = 'animation1',
+          SECOND_ANIMATION_LABEL = 'animation2';
+    let sprite, projectAnimations;
+
+    beforeEach(function () {
+      sprite = createSprite(0, 0);
+
+      // We manually preload animations onto p5.projectAnimations for the use of
+      // setAnimation.
+      projectAnimations = {
+        [ANIMATION_LABEL]: createTestAnimation(8),
+        [SECOND_ANIMATION_LABEL]: createTestAnimation(10)
+      };
+      gameLabP5.p5.projectAnimations = projectAnimations;
+    });
+
+    it('throws if the named animation is not found in the project', function () {
+      expect(() => {
+        sprite.setAnimation('fakeAnimation');
+      }).to.throw(`Unable to find an animation named "fakeAnimation".  Please make sure the animation exists.`);
+    });
+
+    it('makes the named animaiton the current animation, if the animation is found', function () {
+      sprite.setAnimation(ANIMATION_LABEL);
+
+      // Current animation label should be animation label
+      expect(sprite.getAnimationLabel()).to.equal(ANIMATION_LABEL);
+
+      // Current animation will be a clone of the project animation:
+      expectAnimationsAreClones(sprite.animation, projectAnimations[ANIMATION_LABEL]);
+    });
+
+    it('changes the animation to first frame and plays it by default', function () {
+      sprite.setAnimation(ANIMATION_LABEL);
+
+      // Animation is at frame 1
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      // Animation is playing
+      expect(sprite.animation.playing).to.be.true;
+    });
+
+    it('changing to a new animation resets to first frame and starts playing', function () {
+      // Set first animation, advance a few frames
+      sprite.setAnimation(ANIMATION_LABEL);
+      expect(sprite.animation.getFrame()).to.equal(0);
+      expect(sprite.animation.playing).to.be.true;
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      expect(sprite.animation.getFrame()).to.equal(3);
+      sprite.pause();
+      expect(sprite.animation.playing).to.be.false;
+
+      // Set second animation, observe that we are back on first frame
+      sprite.setAnimation(SECOND_ANIMATION_LABEL);
+      expect(sprite.animation.getFrame()).to.equal(0);
+      expect(sprite.animation.playing).to.be.true;
+    });
+
+    it('calling setAnimation with the current animation does not reset frame or play state', function () {
+      // Set first animation, advance a few frames
+      sprite.setAnimation(ANIMATION_LABEL);
+      expect(sprite.animation.getFrame()).to.equal(0);
+      expect(sprite.animation.playing).to.be.true;
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      expect(sprite.animation.getFrame()).to.equal(3);
+      sprite.pause();
+      expect(sprite.animation.playing).to.be.false;
+
+      // Set second animation, observe that we are back on first frame
```

Comment:
I think this comment is probably copied from above and is irrelevant. Also isn't this the case where we would want the animation to be playing, and continue playing but not reset? Wouldn't be a bad idea to test both cases.